### DISCUSSION TITLE
Fuldho 3.5" Screen

### DISCUSSION THREAD

**Discussion by ghost** [mathoudebine/turing-smart-screen-python#298] [Ideas]
I noticed this on the main README and decided to investigate the hardware. I've spent the last week or so reverse engineering it. I am not quite finished but I am getting close.

The hardware works off a completely different approach to other more common screens. It doesn't use the regular approach of implementing a custom protocol over data endpoints. Instead it pretends to be a USB HID and uses vendor specific reports.

It doesn't allow arbitrary drawing via BLIT type operations. The screen does all the drawing, using a "theme" that was previously uploaded by the theme editor. In normal operation, custom reports are used to update the system information on the screen. 

I am currently working on understanding how the "theme" binary file is formatted, which is the last major thing left before alternative software can be written.

Given all this, is there any room for this device in this project? Given what I know of the project, chances are the best you could do is translate the existing themes from the current editor and add something to pass system data to the screen.

**Comment by mathoudebine** [comment]
Thanks for investigating on the Fuldho screen! This is interesting to see a smart screen working differently than the other models I've seen so far.

I think it would be difficult to integrate this screen to the existing project because of the current code architecture: the https://github.com/mathoudebine/turing-smart-screen-python/blob/main/library/lcd/lcd_comm.py abstract class that is used as the base for all supported display drivers (Turing / Xuanfang..) has been written so that the display does not actually know any of the hardware sensors data. Instead it is only used to draw generic content on the display like text, progress bars or bitmaps.

If the Fuldho screen needs to know which sensor the data comes from to display it properly, it would require to change the base architecture of this project.

There may be a way to do it that I didn't think of! For now I will just add a link to your project https://github.com/braewoods/hidss in the Readme for other people to know

**Reply by matu6968** [reply]
the following repo listed is no longer available due to a account deletion by the original poster, remove any references about that repo and have support for it in the future.

**Comment by ghost** [comment]
Technically it doesn't really care where it comes from. It just has some input data IDs set aside for certain purposes that it then updates the screen with if the currently loaded theme uses them at all. I think it may be possible to add new features the original software does not support in terms of theme generation.

**Comment by heavyoak** [comment]
hi, since hidss is gone? is there a different alt software to control the Fuldho 3.5 screen? the official software is buggy and on oping disabled my UAC settings which was a huge red flag.

**Reply by EmilioHernandezZ** [reply]
It did the same thing to me, but from there I put it back where it should be and that's it, it hasn't given me an error since I put it back the way it was.

**Comment by ByWallkroft** [comment]
Hi. I make a big mistake an get this 3.5 screen. Haw anybody got alternative software for this piece of sh... khm. 
I got too mush bugs with original software just like disconnecting every 0.5s at Win11 (changing user will help) or it's didn't see my rtx5070 sensors, just like example 
And I so sorry, my english is... Well... This screen is better then my english)))

**Reply by Kae1thas** [reply]
Hey, have you found a solution to the problem with this piece of screen?

**Reply by ByWallkroft** [reply]
> Hey, have you found a solution to the problem with this piece of screen?

Nope(

**Reply by GeorgeVasilev2005** [reply]
fuckin irresponsible сhinese people... this screen is my birthday present for my close friend. Oh Lord, I wish I had read about the problems with this display earlier. Now my friend the programmer is fucking with this display. We are thinking about how to rewrite the code so that it supports the 50 series of nVIDIA. If anyone has a lot of free time and energy, please respond, and we will work together not for money, but for the right of other people who bought this piece of shit to make it work properly

**Comment by mathoudebine** [comment]
For people that want to use the Fuldho screen, a fork is available here https://github.com/Agentry433/turing-smart-screen-python-HIDdev

**Reply by ByWallkroft** [reply]
> For people that want to use the Fuldho screen, a fork is available here https://github.com/Agentry433/turing-smart-screen-python-HIDdev

Deleted((( Have you alt link for it?

**Reply by LessPlum3393** [reply]
https://github.com/Agentry433/turing-smart-screen-python-HIDdev2

**Reply by ByWallkroft** [reply]
> For people that want to use the Fuldho screen, a fork is available here https://github.com/Agentry433/turing-smart-screen-python-HIDdev

And how it use? I kinda retarded

**Reply by ByWallkroft** [reply]
> For people that want to use the Fuldho screen, a fork is available here https://github.com/Agentry433/turing-smart-screen-python-HIDdev

That for linux only(

**Reply by LessPlum3393** [reply]
Yes, it's currently Linux-only. I opened an issue requesting Windows support, so hopefully the developer will consider adding it. We'll see what they decide.

**Comment by braewoods** [comment]
I'm the original developer of that experimental `hidss`. It's been awhile since I looked around here. I archived that source code over at GitLab. See here: https://gitlab.com/braewoods/usb-smart-screen/-/tree/37c4b21466085090f1deda7b445a3a217723ec7e/

It's not finished, but I intend to return to finishing support for all 3 screen variants from the original vendor. Then I found I hit a dead-end so I need a new design to move forward. Though right now I am busy getting my hardware in order before I can return to active development on anything. Perhaps next year I can return to it all.
